src/MSBuild/XMake.cs

```diff
@@ -627,7 +627,7 @@ string commandLine
 
                     // Honor the low priority flag, we place our selves below normal
                     // priority and let sub processes inherit that priority.
-                    ProcessPriorityClass priority = lowPriority ? ProcessPriorityClass.BelowNormal : ProcessPriorityClass.Normal;
+                    ProcessPriorityClass priority = lowPriority ? ProcessPriorityClass.BelowNormal : Process.GetCurrentProcess().PriorityClass;
                     Process.GetCurrentProcess().PriorityClass = priority;
```

Comment:
We really shouldn't touch priority unless the lowPriority flag was passed... it  just makes the code ripe for these kinds of bugs.